### DISCUSSION TITLE
Another error when trying to build.

### DISCUSSION THREAD

**Discussion by AkshaySC1188** [lencx/ChatGPT#182] [Q&A]
Ok...I thought i got it working but i am getting an error like below.

Error A public key has been found, but no private key. Make sure to set `TAURI_PRIVATE_KEY` environment variable.error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

**Comment by lencx** [comment]
I answered this question here, just turn off the configuration. https://github.com/lencx/ChatGPT/issues/123#issuecomment-1368566083

**Reply by AkshaySC1188** [reply]
I tried doing that. but now i am getting these two errors..

error[E0432]: unresolved import `tauri::updater`
  --> src\utils.rs:11:12
   |
11 | use tauri::updater::UpdateResponse;
   |            ^^^^^^^ could not find `updater` in `tauri`

and 

error[E0599]: no method named `updater` found for struct `AppHandle` in the current scope
   --> src\utils.rs:136:26
    |
136 |         let result = app.updater().check().await;
    |                          ^^^^^^^ method not found in `AppHandle`

**Reply by lencx** [reply]
Sorry, the previous method is not working due to source code change, you can generate signature locally to replace `pubkey` and rebuild it. Here is the signature generation documentation https://tauri.app/v1/guides/distribution/updater/#signing-updates

Step1: The password is optional, if not, it does not need to be set

```bash
# macOS set environment variables：
export TAURI_PRIVATE_KEY="********"
export TAURI_KEY_PASSWORD="********"

# Windows set environment variables：
set TAURI_PRIVATE_KEY="********"
set TAURI_KEY_PASSWORD="********"
```

Step2: Run the dev or build command

```bash
# yarn dev
yarn build
```
